include/nanorange/algorithm/unique.hpp

```diff
@@ -18,30 +19,30 @@ namespace detail {
 struct unique_fn {
 private:
     template <typename I, typename S, typename R, typename Proj>
-    static constexpr I impl(I first, S last, R& comp, Proj& proj)
+    static constexpr subrange<I> impl(I first, S last, R& comp, Proj& proj)
     {
-        first = adjacent_find_fn::impl(std::move(first), last, comp, proj);
+        I it = adjacent_find_fn::impl(first, last, comp, proj);
 
-        if (first == last) {
-            return first;
+        if (it == last) {
+            return {std::move(it), std::move(nano::next(first, last))};
```

Comment:
I think you can say `return {it, std::move(it)};` here (curly braces guarantee left-to-right evaluation).